blueocean-dist/pom.xml

```diff
@@ -0,0 +1,73 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
> Or will it have other uses?

That's what I was wondering when choosing dist, but I don't find any other use right now, and given we will not publish the artifact, it'll not be an issue the rename it latter if needed. So, I'm also in favor pick up a more explicit name. `blueocean-docker-dev-image` could be a good feet, but maybe getting rid of this module would be even better?

I created this module to cleanly separate our 2  Dockerfile, but if we choose to move #403 to a separate repository, we could move this back into `blueocean-plugin/pom.xml` and a Dockerfile in the root folder.